DEVELOPMENT.md

```diff
@@ -429,12 +406,71 @@ git submodule update --init --remote
 git submodule foreach "git fetch --all"
 ```
 
-Then run tests as usual: `cargo test`.
+Then run tests as usual: `cargo test --all-features --all-targets`.
+
+We strive for two kind of tests, the classic unit tests inside of the
+implementation files. Additionally we have a suit of integration tests esp. to
+assert the API the proxy provides from a high-level consumer perspective, these
+can be found under `proxy/tests`. To find out where to place and how to layout
+tests check Rust book [test chapter][rt].
+
+### File structure
+
+We understand the GraphQL API as one potential surface to expose the
+Application domain logic. Therefore we try to treat it as a thin layer exposing
+well-typed entities. The heavy lifting is done in the modules named after the
+protocols we consume - `CoCo` for code collaboration and `Registry` for global
+unique entries for users, projects and organisations. By isolating concerns
+this way, we hope to enable ease-of-contribution to downstream teams.
+Empowering them to reflect changes in their public APIs easily with code
+contributions to Upstream.
+
+```
+proxy/src/
+├── coco.rs
+├── env.rs
+├── error.rs
+├── graphql
+│   ├── api.rs
+│   ├── error.rs
+│   ├── project.rs
+│   └── schema.rs
+├── graphql.rs
+├── lib.rs
+├── main.rs
+└── registry.rs
+```
 
 💡 *You'll have to run the submodule commands every time there are any updates
 to the test fixture repository.*
 
 
+## Scripts
+
+To get a list of all available script commands, run: `yarn run`.
+Here is a list of the most commonly used ones:
+
+```sh
+yarn start            # Start Upstream in development mode
+
+yarn test             # Run Cypress end-to-end tests
+yarn test:debug       # Show the Cypress GUI, handy for visual debugging
+
+yarn dist             # Bundles Upstream into an installable package
+
+yarn generate:colors  # Update color CSS variables in public/colors.css from
+                      # colors.json
+
+yarn graphiql         # Open GraphiQL with explorer extension in browser
+
+yarn release          # Start a two-step process to cut a new release, for more
+                      # details have a look at ../DEVELOPMENT.md
+
+yarn prettier:check   # Check UI code formatting
+yarn prettier:write   # Auto-format UI code
+yarn lint             # Check UI code for linting errors
+```
```

Comment:
We also have a bunch of helper scripts for the proxy in our `package.json`, should we maybe include those here too with proper descriptions?
```
proxy:build          # cd proxy && cargo build --all-features --all-targets
proxy:build:release  # cd proxy && cargo build --release --all-features --all-targets
proxy:build:test     # cd proxy && cargo test --no-run
proxy:start          # cd proxy && cargo run --release -- --registry=emulator
proxy:start:test     # cd proxy && cargo run -- --registry=emulator --test
```

Reply:
Generally I find these outputs of cli options to be prone to quickly get outdated. I'd rather show a command which shows all the commands.

Reply:
I know what you mean. This section expands a bit on the commands however. It's meant to explain when they should be used. As the `yarn run` command can only show the commands themselves, but there is no way to inline documentation in `package.json`.

I just provided the `# cd proxy && cargo run` comments as a reference, the idea was that you'd write the explanation in a comment...

Reply:
Addressed in 4b037b7